system/jlib/jtrace.hpp

```diff
@@ -55,6 +231,145 @@ class jlib_decl LogTrace
     void setLocalId(const char* id);
 };
 
+/**
+ * @brief This follows open telemetry's span attribute naming conventions
+ *  Known HPCC span Keys could be added here
+ *  Specialized span keys can also be defined within the scope of a span 
+ */
+namespace HPCCSemanticConventions
+{
+static constexpr const char *kGLOBALIDHTTPHeader = "HPCC-Global-Id";
+static constexpr const char *kCallerIdHTTPHeader = "HPCC-Caller-Id";
+}
+
+// github copilot generated comment:
+// This class provides a high-level interface for managing tracing and profiling information in the HPCC Systems platform.
+// The TraceManager class is responsible for creating and managing instances of the Tracer class, which is used to instrument code for tracing and profiling. The Tracer class provides methods for starting and ending spans, adding attributes to spans, and propagating trace context across different services and systems.
+// Overall, the TraceManager class provides a convenient and flexible way to instrument code for tracing and profiling in the HPCC Systems platform. By using the Tracer class and the TraceManager class, developers can gain insight into the performance and behavior of their applications and diagnose issues in distributed systems.
+class TraceManager
+{
+private:
+    //Used as the opentel trace name, refered to as name of library being instrumented
+    const char * tracerName = nullptr;
+
+public:
+    TraceManager(const char * moduleName)
+    {
+        //InitModuleObjects();
+        initTracer(); //Still not sure where this should be done,
+                      //but it needs to be done once per process before any tracers/spans are created
+        tracerName = moduleName; //1 tracer for each module/library being instrumented
+    };
+
+    TraceManager(const std::string & moduleName)
+    {
+        TraceManager(moduleName.c_str());
```

Comment:
This is almost certainly going to lead to traceName pointing at a name that has been fred.